Sources/Serialization/JSONSerialization.swift

```diff
@@ -0,0 +1,101 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+import PackageDescription
+
+/// A JSON representation of an element.
+public protocol JSONSerializable {
+    
+    /// Return a JSON representation.
+    func toJSON() -> AnyObject
+}
+
+public func jsonString(package: PackageDescription.Package) throws -> String {
+    
+    let json: AnyObject = package.toJSON()
+    let data = try NSJSONSerialization.data(withJSONObject: json, options: .prettyPrinted)
+    guard let string = String(data: data, encoding: NSUTF8StringEncoding) else { fatalError("NSJSONSerialization emitted invalid data") }
+    return string
+}
+
+extension NSMutableDictionary {
+    static func withNew(block: @noescape (dict: NSMutableDictionary) -> ()) -> NSMutableDictionary {
+        let dict = NSMutableDictionary()
+        block(dict: dict)
+        return dict
+    }
+}
+
+extension SystemPackageProvider: JSONSerializable {
+    public func toJSON() -> AnyObject {
+        let (name, value) = nameValue
+        
+        return NSMutableDictionary.withNew { (dict) in
```

Comment:
I see what you mean, on the other hand there are changes being implemented that disallow things like `String -> String` and instead force `(String) -> String`. This case feels like the new, enforced style, just without the return value.

I think it's completely valid to want to change this, but maybe we should change the default in Xcode, because this is what you get at the moment. Usually I need a very good reason to fight Xcode's formatting on something.